arrow-array/src/builder/fixed_size_list_builder.rs

```diff
@@ -247,6 +298,180 @@ mod tests {
         assert_eq!(3, list_array.value_length());
     }
 
+    #[test]
+    fn test_fixed_size_list_array_builder_with_field() {
+        let values_builder = Int32Builder::new();
+        let mut builder = FixedSizeListBuilder::new(values_builder, 3).with_field(Field::new(
+            "list_element",
+            DataType::Int32,
+            true,
+        ));
+
+        //  [[0, 1, 2], null, [3, null, 5], [6, 7, null]]
+        builder.values().append_value(0);
+        builder.values().append_value(1);
+        builder.values().append_value(2);
+        builder.append(true);
+        builder.values().append_null();
+        builder.values().append_null();
+        builder.values().append_null();
+        builder.append(false);
+        builder.values().append_value(3);
+        builder.values().append_null();
+        builder.values().append_value(5);
+        builder.append(true);
+        builder.values().append_value(6);
+        builder.values().append_value(7);
+        builder.values().append_null();
+        builder.append(true);
```

Comment:
It might be nice to avoid repeating this same setup multiple times in the tests (and it would make it clear the tests share the same data)

for example
```rust
fn make_list_builder() -> FixedSizedListBuilder<i32> { 
  ...
}
```


```suggestion
        let builder = make_list_builder();
```

Reply:
Makes sense, I'll go through the tests and rework the builders.